src/cmdline/mod.rs

```diff
@@ -223,7 +209,8 @@ impl Cmdline {
         self.insert_str(kv_str)
     }
 
-    /// Validates and inserts a string to the end of the current command line.
+    /// Validates and inserts a string representing boot args
```

Comment:
I think this doc comment is a bit misleading. The purpose of the function is to "insert a string in the boot args". And then we can say: "Returns an error if the string is invalid". This was wrong before as well, but since we are updating it, we can also correct it. With the current wording it looks like the function does more than one thing which is not really the case.